src/main/scala/io/iohk/ethereum/blockchain/sync/SyncStateSchedulerActor.scala

```diff
@@ -202,11 +509,225 @@ object SyncStateSchedulerActor {
   case object StateSyncFinished extends SyncStateSchedulerActorResponse
   case object WaitingForNewTargetBlock extends SyncStateSchedulerActorResponse
 
-  case class StateSyncStats(saved: Long, missing: Long)
-
   case class GetMissingNodes(nodesToGet: List[ByteString])
   case class MissingNodes(missingNodes: List[SyncResponse], downloaderCapacity: Int)
 
   case class BloomFilterResult(res: BloomFilterLoadingResult)
 
+  sealed trait RequestResult
+  case class RequestData(nodeData: NodeData, from: Peer) extends RequestResult
+  case class RequestFailed(from: Peer, reason: String) extends RequestResult
+
+  sealed trait ProcessingError
+  case class Critical(er: CriticalError) extends ProcessingError
```

Comment:
What is the difference between `Critical` and `DownloaderError` with `critical = true`?

Reply:
Criticial stop the sync entierly as the trie is malformed for some reason, and `DownloaderError ` wihth critical only blacklist peer. I will change the naming here as it can be confusing